src/labs.txt

```diff
@@ -2809,6 +2809,216 @@ Output:
 =hist
 EOF
 
+----------------------------------------------------------------------
+h1. Finding Commits That Introduced a Bug
+
+h2. Goals
+
+* Learn how to use @git bisect@ to find a commit that introduced a change.
+
+p. Bugs happen, and it can be necessary to find the cause of a bug without
+knowing what code specifically is involved. Git allows you to efficiently
+search for a commit as long as you can describe your search criteria in terms
```

Comment:
Even without an executable test, it's still more efficient / helpful to use bisect to keep track of which commits were good and which commits were bad within a range.

Reply:
I'm not sure what you're getting at here.

Reply:
"Git allows you to efficiently search for a commit as long as you can describe your search criteria in terms of a test that can be executed."

"A test that can be executed" implies to me that you need some kind of runnable command to perform the test.  This sentence _could_ cover the "I'm manually refreshing my browser to see if the behavior is better now" scenario too, but I think we can reword to cover it more clearly.

Reply:
What about: "Git allows you to efficiently search for a commit by taking some action and telling git if the commit is closer or farther than the one you are looking for."

Reply:
"closer or farther" isn't really accurate -- you never really get further away from the target commit as you proceed.  It's more "before or after the change you are looking for", but that's a little awkward as an intro too.  It's hard to describe abstractly.  I feel like it is easier to clearly describe bisect's behavior specifically in the context of a bug search.

```
Bugs happen, and it can be necessary to find the cause of a bug without
knowing what code specifically is involved. Git allows you to efficiently
search a series of commits to find the commit that introduced the bug.

Git bisect asks you to declare a known good commit and a known bad commit.  Then git bisect will pick a commit in the middle of that set and ask you test that commit.  You'll test your project at that point either manually or via an automated test, and then you'll mark the commit as good or bad.  Git bisect will continue to pick the midpoint between the latest good and bad commits until you land on the exact commit that introduced the bug.
```

Reply:
@joealba Updated. Preview should be up to date shortly.